constraints.pro

```diff
@@ -0,0 +1,67 @@
+% Enforces that all workspaces depend on other workspaces using `workspace:*` in devDependencies
+gen_enforced_dependency(WorkspaceCwd, DependencyIdent, 'workspace:*', 'devDependencies') :-
+  workspace_has_dependency(WorkspaceCwd, DependencyIdent, DependencyRange, 'devDependencies'),
+  % Only consider dependency ranges that start with 'workspace:'
+  atom_concat('workspace:', _, DependencyRange).
+
+% Enforces the license in all public workspaces while removing it from private workspaces
+gen_enforced_field(WorkspaceCwd, 'license', 'MIT') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
+gen_enforced_field(WorkspaceCwd, 'license', null) :-
+  workspace_field(WorkspaceCwd, 'private', true).
+
+% Enforces the repository field for all public workspaces while removing it from private workspaces
+gen_enforced_field(WorkspaceCwd, 'repository.type', 'git') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
+gen_enforced_field(WorkspaceCwd, 'repository.url', 'https://github.com/babel/babel.git') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
+gen_enforced_field(WorkspaceCwd, 'repository.directory', WorkspaceCwd) :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
+gen_enforced_field(WorkspaceCwd, 'repository', null) :-
+  workspace_field(WorkspaceCwd, 'private', true).
+
+% Enforces 'publishConfig.access' is set to public for public workspaces while removing it from private workspaces
+gen_enforced_field(WorkspaceCwd, 'publishConfig.access', 'public') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
+gen_enforced_field(WorkspaceCwd, 'publishConfig.access', null) :-
+  workspace_field(WorkspaceCwd, 'private', true).
+
+% Enforces the engines.node field for all workspaces except '@babel/eslint*'
+gen_enforced_field(WorkspaceCwd, 'engines.node', '>=6.9.0') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true),
+  % Get the workspace name
+  workspace_ident(WorkspaceCwd, WorkspaceIdent),
+  % Exempt from the rule as it supports '>=4'. TODO: remove with the next major
+  WorkspaceIdent \= '@babel/plugin-proposal-unicode-property-regex',
+  % Exempt from the rule as it supports '>=6.0.0'. TODO: remove with the next major
+  WorkspaceIdent \= '@babel/parser',
```

Comment:
Are these two actually built for `>=6.9.0` but the `engines.node` value wasn't updated or are they correct and special cased?

Reply:
Oh good catch, they are both built for `6.9.0`.

Reply:
Should I let them get updated or wait for the next major to be on the safe side?

Reply:
Lets keep them for now, just to be safe.